### PR TITLE
A few small tweaks to setkey() docs

### PR DESCRIPTION
**Pull request by hadley**
* Created new "good practice" section
* Broke good practice up into multiple paragraphs
* Added spaced between why and \code{setkey}

I think this makes the docs a little easier to read, but please let me know if this sort of PR isn't wanted.

### CODE REVIEW

File: man/setkey.Rd [hunk 1]
```diff
@@ -63,16 +63,21 @@ counting sort and forwards radix. We also avoid hash tables.
...
 If you use column numbers then bugs (possibly silent) can more easily creep into
 your code as time progresses if changes are made elsewhere in your code; e.g., if
 you add, remove or reorder columns in a few months time, a \code{setkey} by column
 number will then refer to a different column, possibly returning incorrect results
 with no warning. (A similar concept exists in SQL, where \code{"select * from \dots"} is considered poor programming style when a robust, maintainable system is
-required.)  If you really wish to use column numbers, it is possible but
+required.)
+
+If you really wish to use column numbers, it is possible but
 deliberately a little harder; e.g., \code{setkeyv(DT,colnames(DT)[1:2])}.
```
Comment (MichaelChirico) [inline man/setkey.Rd]: while you're in here could you change colnames to just names? IINM names would be more canonical

Comment (hadley) [reply man/setkey.Rd]: Done!

**Comment by MichaelChirico** [comment]
LGTM, thanks!

**Comment by mattdowle** [comment]
Many thanks, @hadley! Yes PRs like this are much appreciated.

**Pull request merged**